### DISCUSSION TITLE
moving branches between different wikis

### DISCUSSION THREAD

**Discussion by dalcolmo** [zim-desktop-wiki/zim-desktop-wiki#2602] [Ideas]
I have often the need to move whole branches of a Zim notebook to another Zim notebook. Unfortunately that is not so easy. Of course, it breaks links, particularly links to attachments when the document root differs between different notebooks.

I use a wiki per project and store such wikis not under ~/Notebooks, but in the project folder. More importantly, the root directory of the notebook refers then to the root of the project and is therefore different for each project.

I also use a couple of additional wikis: one to keep track of all my work to-do's and one for my private stuff.

Cutting and pasting whole branches between notebooks does not seem to work.

I tried then to set up a "master" notebook in a parent folder of all the other notebooks, and tried to move pages than within that master notebook. That also has does not work properly, because the links to root pages in the "children" notebooks are still interpreted as links to the root in the master notebook, but that is of course the wrong link now and either shows up as a link to a non-existing page, or worse, links to the wrong page.  If the children notebooks have different document roots, the problem is worse.

Of course, there is now way to prevent breakage for links moved to a new notebook, when the target remains in the old notebook.

I was using version 0.74.3 on Ubuntu LTS until today. I see there is a mention of  "sub-notebooks" in the changelog of 0.75.1 Perhaps I am missing this feature?
I just upgraded to Issue 0.75.2.

**Comment by introt** [comment]
> Of course, there is now way to prevent breakage for links moved to a new notebook, when the target remains in the old notebook.

You can use interwiki links, see https://zim-wiki.org/manual/Help/Links.html

> I tried then to set up a "master" notebook in a parent folder of all the other notebooks, and tried to move pages than within that master notebook. That also has does not work properly, because the links to root pages in the "children" notebooks are still interpreted as links to the root in the master notebook, but that is of course the wrong link now and either shows up as a link to a non-existing page, or worse, links to the wrong page. If the children notebooks have different document roots, the problem is worse.

You might be able to work around this using `find` and `sed -i`, but really I'd question the need for separate wikis if this problem comes up often enough to matter - there's a plugin for limiting the index to a smaller namespace if that's why you've resorted to using multiple. If the problem is slow search, you might find that the new full text search is a lot faster.

**Comment by dalcolmo** [comment]
The latest implementation of the tasklist is really much better. Perhaps I will switch back to using one single wiki for day-to-day work.
